source/val/validate_decorations.cpp

```diff
@@ -917,6 +926,86 @@ spv_result_t CheckDecorationsOfBuffers(ValidationState_t& vstate) {
   return SPV_SUCCESS;
 }
 
+spv_result_t CheckDecorationsCompatibility(ValidationState_t& vstate) {
+  static const std::unordered_set<SpvDecoration, SpvDecorationHash>
+      kAtMostOncePerID = {
+          SpvDecorationArrayStride,
+      };
+  static const std::unordered_set<SpvDecoration, SpvDecorationHash>
+      kAtMostOncePerMember = {
+          SpvDecorationOffset,
+          SpvDecorationMatrixStride,
+          SpvDecorationRowMajor,
+          SpvDecorationColMajor,
+      };
+  static const std::vector<std::unordered_set<SpvDecoration, SpvDecorationHash>>
+      kMutuallyExclusivePerMember = {
+          {SpvDecorationRowMajor, SpvDecorationColMajor},
```

Comment:
I would add BufferBlock and Block here too although they are not member decorations.

Reply:
Done